File: src/net/theharrisoncrafter/Commands/Troll.java
package net.theharrisoncrafter.Commands;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class Troll implements CommandExecutor{
	public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args){	
		
		if(sender instanceof Player){
			if(sender.hasPermission("GodCraft.Troll")){
				Player targetPlayer;
				
				if(args.length > 1){
					return false;
				}
				else if(args.length < 1){
					targetPlayer = (Player) sender;
				}
				else if(args.length == 1){
					targetPlayer = sender.getServer().getPlayer(args[0]);
				}
				else{
					targetPlayer = null;
				}
				
				if(targetPlayer == null){
					sender.sendMessage(ChatColor.RED + "That player is not online!");
					return true;
				}		
							
				targetPlayer.getWorld().createExplosion(targetPlayer.getLocation(), 0L);
				targetPlayer.getWorld().strikeLightningEffect(targetPlayer.getLocation());			
				return true;
			}
			else{
				sender.sendMessage(ChatColor.RED + "You do not have permissions to do that!");
				return true;
			}
		}
		else{
			Player targetPlayer;
			if(args.length < 1 || args.length > 1){
				return false;
			}
			
			targetPlayer = Bukkit.getPlayer(args[0]);
			
			if(targetPlayer == null){
				sender.sendMessage(ChatColor.RED + "That player is not online!");
				return true;
			}		
						
			targetPlayer.getWorld().createExplosion(targetPlayer.getLocation(), 0L);
			targetPlayer.getWorld().strikeLightningEffect(targetPlayer.getLocation());			
			return true;
		}
	}
}

File: src/main/kotlin/co/orre/godcraft/Commands/Troll.kt
package co.orre.godcraft.Commands

import co.orre.godcraft.God
import org.bukkit.Bukkit
import org.bukkit.ChatColor as CC
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class Troll(val plugin: God) : CommandExecutor {
    override fun onCommand(sender: CommandSender, cmd: Command, commandLabel: String, args: Array<String>): Boolean {
        if (sender is Player) {
            if (sender.hasPermission("godcraft.troll")) {
                val targetPlayer: Player?

                when {
                    args.size > 1 -> return false
                    args.isEmpty() -> targetPlayer = sender
                    args.size == 1 -> targetPlayer = Bukkit.getPlayer(args[0])
                    else -> targetPlayer = null
                }

                if (targetPlayer == null) {
                    sender.sendMessage("${CC.RED}That player could not be found!")
                    return true
                }

                targetPlayer.world.createExplosion(targetPlayer.location, 0f)
                targetPlayer.world.strikeLightningEffect(targetPlayer.location)
                return true
            }
            sender.sendMessage("${CC.RED}You do not have permissions to do that!")
            return true
        }
        if (args.size != 1) return false
        val targetPlayer: Player? = Bukkit.getPlayer(args[0])
        if (targetPlayer == null) {
            sender.sendMessage("${CC.RED}That player could not be found!")
            return true
        }

        targetPlayer.world.createExplosion(targetPlayer.location, 0f)
        targetPlayer.world.strikeLightningEffect(targetPlayer.location)
        return true
    }
}
